### PR TITLE
fix/3804-part2

### LINKED ISSUE #3804: email templates: make them more flexible
Related to #3664 

Currently, email templates use a static list of variables. If you use Root Portal Modules and you add new mandatory fields, it would be nice to have them automatically integrated in the resulting emails.

### Example

You add a "company" mandatory field for a sponsor-based registration in a Root Portal Module, you will see this field and associated value in resulting emails.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Improve email templates flexibility.

# Impacts
Email/Sponser email templates

# Issue
fixes #3804

# Delete branch after merge
(REQUIRED)
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Automatically display mandatory fields in email/sponsor activation emails.

### CODE REVIEW

**Comment by extrafu** [comment]
@cgx @fdurand @nqb Please act quickly on this one to get it in for v12 once for all. Thanks!

File: html/captive-portal/templates/emails/emails-guest_email_activation.html [hunk 1]
```diff
@@ -41,35 +41,40 @@
...
+
   </style>
   <style media="screen and (min-width:480px)">
     .moz-text-html .mj-column-per-100 {
       width: 100% !important;
       max-width: 100%;
     }
+
+  </style>
+  <style type="text/css">
```
Comment (fdurand) [inline html/captive-portal/templates/emails/emails-guest_email_activation.html:77]: we have to add it twice ?

File: html/captive-portal/templates/emails/emails-guest_sponsor_activation.html [hunk 2]
```diff
@@ -80,6 +82,9 @@
       width: 50% !important;
       max-width: 50%;
     }
+
+  </style>
+  <style type="text/css">
```
Comment (fdurand) [inline html/captive-portal/templates/emails/emails-guest_sponsor_activation.html:87]: duplicated

File: html/captive-portal/templates/emails/emails-guest_email_activation.html (hunk 1 above)
Comment (jrouzierinverse) [reply html/captive-portal/templates/emails/emails-guest_email_activation.html:77]: This is coming from mjml after processing.

File: html/captive-portal/templates/emails/emails-guest_sponsor_activation.html (hunk 2 above)
Comment (jrouzierinverse) [reply html/captive-portal/templates/emails/emails-guest_sponsor_activation.html:87]: This is coming from mjml after processing.

**Review by fdurand — APPROVED** [review]

**Review by nqb — APPROVED** [review]
I did a short review because I know @fdurand tested this feature carefully.

**Pull request merged**